libraries/AP_Motors/AP_MotorsHeli_Swash.cpp

```diff
@@ -243,12 +244,10 @@ void AP_MotorsHeli_Swash::calculate(float roll, float pitch, float collective)
 // set_linear_servo_out - sets swashplate servo output to be linear
 float AP_MotorsHeli_Swash::get_linear_servo_output(float input) const
 {
-
     input = constrain_float(input, -1.0f, 1.0f);
 
-    //servo output is calculated by normalizing input to 50 deg arm rotation as full input for a linear throw
-    return safe_asin(0.766044f * input) * 1.145916;
-
+    // servo output is calculated by normalizing input to max servo horn rotation angle
+    return safe_asin(sinf(radians(_linear_swash_servo_ang_deg.get())) * input) * 1/radians(_linear_swash_servo_ang_deg.get());
```

Comment:
```suggestion
   const linearisation_angle = radians(get_lin_angle());
    return safe_asin(sinf(linearisation_angle ) * input) * 1/linearisation_angle;
```
Just to save doing the radians conversion twice.